### PR TITLE
Migrate tests to use JUnit4

### PR DESCRIPTION
**Pull request by phillipm**
I got tired of J2MEs dated version of JUnit.

This enables easier command-line and IDE test invocations.

This creates some problems with TextFormTests and XPathParseTest that I will fix as soon as this gets merged.

This will require adding JUnit4 as a lib dependency in Android Studio.

**Shares commits with https://github.com/dimagi/javarosa/pull/108.**

### CODE REVIEW

File: core/src/test/java/org/javarosa/xpath/test/XPathParseTest.java [hunk 1]
```diff
@@ -209,34 +191,26 @@
...
-
-    public XPathParseTest() {
-        super();
-    }
-
-    public Test suite() {
-        TestSuite aSuite = new TestSuite();
+    // TODO PLM: fix this!
+    /*
+    public static TestSuite suite() {
```
Comment (phillipm) [inline core/src/test/java/org/javarosa/xpath/test/XPathParseTest.java]: Note that I need to fix this soon.

File: core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java [hunk 2]
```diff
@@ -17,13 +17,11 @@
...
-import j2meunit.framework.TestCase;
-import j2meunit.framework.TestMethod;
-import j2meunit.framework.TestSuite;
-
-public class TextFormTests extends TestCase {
+import org.junit.Test;
+import org.junit.Before;
+import static org.junit.Assert.*;
 
+public class TextFormTests {
```
Comment (phillipm) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: The tests in this file were never being called! Enabling them, it looks like 3 of them are failing

File: core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java [hunk 3]
```diff
@@ -288,6 +229,4 @@ public void testPromptsNoLocalizer() {
             fail("Help text getter/setter broken");
```
Comment (phillipm) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: Fails with a null pointer exception

File: core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java [hunk 4]
```diff
@@ -279,8 +220,8 @@ public void testPromptIDsNoLocalizer() {
         }
```
Comment (phillipm) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: Fails due to equality check in between String and QuestionString objects

File: core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java [hunk 5]
```diff
@@ -155,9 +92,9 @@ public void testTextForms() {
             fail("getLongText() not falling back to default text form correctly");
```
Comment (phillipm) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: Fails, due to test resource input not matching expected behavior

Comment (ctsims) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: Hey,

These are fixed in the 2.22 branch.

File: core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java (hunk 2 above)
Comment (phillipm) [inline core/src/test/java/org/javarosa/core/form/api/test/TextFormTests.java]: Clayton fixed this, nv

**Pull request merged**